lib/matplotlib/testing/conftest.py

```diff
@@ -125,3 +125,39 @@ def test_imshow_xarray(xr):
 
     xr = pytest.importorskip('xarray')
     return xr
+
+
+def _text_placeholders(monkeypatch):
+    from matplotlib.patches import Rectangle
+
+    def patched_get_text_metrics_with_cache(renderer, text, fontprop, ismath, dpi):
+        """
+        Replace ``_get_text_metrics_with_cache`` with fixed results.
+
+        The usual ``renderer.get_text_width_height_descent`` would depend on font
+        metrics; instead the fixed results are based on font size and the length of the
+        string only.
+        """
+        height = fontprop.get_size()
+        width = len(text) * height / 1.618  # Golden ratio for character size.
+        return width, height, 0
+
+    def patched_text_draw(self, renderer):
+        """
+        Replace ``Text.draw`` with a fixed bounding box Rectangle.
+
+        The bounding box corresponds to ``Text.get_window_extent``, which ultimately
+        depends on the above patched ``_get_text_metrics_with_cache``.
+        """
+        if renderer is not None:
+            self._renderer = renderer
+        if not self.get_visible():
+            return
+        if self.get_text() == '':
+            return
+        bbox = self.get_window_extent()
+        Rectangle(bbox.p0, bbox.width, bbox.height, color='grey').draw(renderer)
```

Comment:
```suggestion
        Rectangle(bbox.p0, bbox.width, bbox.height, facecolor='grey').draw(renderer)
```

`color` would also set the edgecolor, which is transparent by default ('none'). The edgecolor induces a line to be drawn around the region and since lines have antialiasing, the edges can become blury. We don't want that. See also https://github.com/matplotlib/matplotlib/issues/29874#issuecomment-2781694549.